### PR TITLE
Add ci and initialize bazel workspace

### PR DESCRIPTION
**Pull request by bianpengyuan**
Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

Mostly copy paste set up from proxy wasm cpp sdk repo: https://github.com/proxy-wasm/proxy-wasm-cpp-sdk. Will need to add license check in later PR after I figured out how to hook istio license check with github action.

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@mandarjog

**Comment by bianpengyuan** [comment]
@kyessenov ptal

**Comment by bianpengyuan** [comment]
@linsun Do you manage the membership of istio-ecosystem org? I cannot add Kuat as a reviewer, could you please help to invite him as a collaborator? Thanks!

**Comment by linsun** [comment]
Kuat is already invited, the status is pending his response.

**Comment by kyessenov** [comment]
Sorry, my invitation has expired. Github sends it to a wrong email address.

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright Istio Authors.
```
Comment (kyessenov) [inline BUILD:1]: Should copyright be Istio Authors or something else? https://github.com/istio-ecosystem/admiral is not attributed to Istio Authors, for example.

File: .bazelversion [hunk 2]
```diff
@@ -0,0 +1 @@
+2.0.0
```
Comment (kyessenov) [inline .bazelversion:1]: This is fairly old by now. Maybe 3.0.0?

File: .github/workflows/cpp.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Format (clang-format)
+      run: |
+        sudo apt-get install clang-format-9
```
Comment (kyessenov) [inline .github/workflows/cpp.yaml:37]: clang-format-10 maybe? It's been out with Debian Buster.

File: BUILD (hunk 1 above)
Comment (bianpengyuan) [reply BUILD:1]: Any suggestion? Looks like admiral does not specify copyright for the code they wrote. I can just remove them as well.

File: .bazelversion (hunk 2 above)
Comment (bianpengyuan) [reply .bazelversion]: Updated to the latest 3.5.0. We'd need to update the version in upstream proxy-wasm-cpp-sdk as well.

File: .github/workflows/cpp.yaml (hunk 3 above)
Comment (bianpengyuan) [reply .github/workflows/cpp.yaml]: Done.

File: BUILD (hunk 1 above)
Comment (kyessenov) [reply BUILD:1]: Maybe remove it for now? I am not sure about this org policy.

Comment (bianpengyuan) [reply BUILD]: Done.

**Review by kyessenov — APPROVED** [review]

**Comment by mandarjog** [comment]
@bianpengyuan @kyessenov for any subsequent PR, we at least want to run make build / make test or equivalent in the extensions subdir

**Comment by kyessenov** [comment]
Can we add a simple helloworld extension as a first example?

**Comment by mandarjog** [comment]
Yes, let’s do that

**Pull request merged**